HCM rainflow count fails when being shipped a named multi indexed load series.

**Describe the bug**

HCM rainflow count fails when being shipped a named multi indexed load series.

**To Reproduce**
Code snippet to reproduce the bug (no screenshot but pasted code)
```python
    vals = np.array([100., -200., 100., -250., 200., 0., 200., -200.]) * (250+6.6)/250 * 1.4
    signal = pd.DataFrame({11: vals, 12: 2*vals, 13: 3*vals, 'load_step': range(len(vals))}).set_index('load_step').stack()
    signal.index.names = ['load_step', 'element_id']
    signal.name = "load"

    E = 206e3    # [MPa] Young's modulus
    K = 3.1148*(1251)**0.897 / (( np.min([0.338, 1033.*1251.**(-1.235)]) )**0.187)
    #K = 2650.5   # [MPa]
    n = 0.187    # [-]
    K_p = 3.5    # [-] (de: Traglastformzahl) K_p = F_plastic / F_yield (3.1.1)

    extended_neuber = NAL.ExtendedNeuber(E, K, n, K_p)

    detector = FKMNonlinearDetector(
        recorder=RFR.FKMNonlinearRecorder(), notch_approximation_law=extended_neuber
    )
    detector.process_hcm_first(signal).process_hcm_second(signal)
```

**Expected result**

Must work without errors

**Observed result**
```
tests/stress/rainflow/test_fkm_nonlinear.py:837: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/pylife/stress/rainflow/fkm_nonlinear.py:131: in process_hcm_first
    samples, flush = self._adjust_samples_and_flush_for_hcm_first_run(samples)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/pylife/stress/rainflow/fkm_nonlinear.py:530: in _adjust_samples_and_flush_for_hcm_first_run
    samples = samples_without_index.set_index(samples.index.names)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/pandas/core/frame.py:4378: in __getitem__
    indexer = self.columns.get_loc(key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index(['load'], dtype='str'), key = 0

    def get_loc(self, key):
        """
        Get integer location, slice or boolean mask for requested label.
    
        Parameters
        ----------
        key : label
            The key to check its location if it is present in the index.
    
        Returns
        -------
        int if unique index, slice if monotonic index, else mask
            Integer location, slice or boolean mask.
    
        See Also
        --------
        Index.get_slice_bound : Calculate slice bound that corresponds to
            given label.
        Index.get_indexer : Computes indexer and mask for new index given
            the current index.
        Index.get_non_unique : Returns indexer and masks for new index given
            the current index.
        Index.get_indexer_for : Returns an indexer even when non-unique.
    
        Examples
        --------
        >>> unique_index = pd.Index(list("abc"))
        >>> unique_index.get_loc("b")
        1
    
        >>> monotonic_index = pd.Index(list("abbc"))
        >>> monotonic_index.get_loc("b")
        slice(1, 3, None)
    
        >>> non_monotonic_index = pd.Index(list("abcb"))
        >>> non_monotonic_index.get_loc("b")
        array([False,  True, False,  True])
        """
        casted_key = self._maybe_cast_indexer(key)
        try:
            return self._engine.get_loc(casted_key)
        except KeyError as err:
            if isinstance(casted_key, slice) or (
                isinstance(casted_key, abc.Iterable)
                and any(isinstance(x, slice) for x in casted_key)
            ):
                raise InvalidIndexError(key) from err
>           raise KeyError(key) from err
E           KeyError: 0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,13 @@
 In this file noteworthy changes of new releases of pyLife are documented since
 2.0.0.
 
+## 2.2.1
+
+### Bug fixes
+
+* Fix crash when HCM algorithm was given a named multiindex load series (#210)
+
+
 ## 2.2.0
 
 ### Breaking changes

--- a/src/pylife/stress/rainflow/fkm_nonlinear.py
+++ b/src/pylife/stress/rainflow/fkm_nonlinear.py
@@ -527,7 +527,7 @@ class FKMNonlinearDetector(RFG.AbstractDetector):
             # increase the load_step index value by one for all samples
             samples_without_index = samples.reset_index()
             samples_without_index.load_step += 1
-            samples = samples_without_index.set_index(samples.index.names)[0]
+            samples = samples_without_index.set_index(samples.index.names).iloc[:, 0]
 
             # prepend the new zero load sample
             samples = pd.concat([first_sample, samples], axis=0)

--- a/tests/stress/rainflow/test_fkm_nonlinear.py
+++ b/tests/stress/rainflow/test_fkm_nonlinear.py
@@ -821,6 +821,7 @@ def test_element_id():
     vals = np.array([100., -200., 100., -250., 200., 0., 200., -200.]) * (250+6.6)/250 * 1.4
     signal = pd.DataFrame({11: vals, 12: 2*vals, 13: 3*vals, 'load_step': range(len(vals))}).set_index('load_step').stack()
     signal.index.names = ['load_step', 'element_id']
+    signal.name = "load"
 
     E = 206e3    # [MPa] Young's modulus
     K = 3.1148*(1251)**0.897 / (( np.min([0.338, 1033.*1251.**(-1.235)]) )**0.187)

